CoqPlugin.py

```diff
@@ -633,6 +631,11 @@ def run(self, edit):
         if "coq" not in self.view.scope_name(0):
             log.write("not inside a coq buffer")
             return
+        for worker in coq_threads.values():
+            response_view = getattr(worker.display, "response_view", None)
+            if response_view is not None and response_view.id() == self.view.id():
```

Comment:
It is both a strength and a weakness of Python that this code works.

While I admire the brevity, I think this code would be better structured by introducing a new method on the `CoqDisplay` class:

```Python
    def owns_view(self, view):
        """This method may only be called from the main thread.

        Returns True iff the argument is a subordinate view owned by this
        display (and therefore not a view into a normal Coq file)."""
        raise NotImplementedError()
```

The `SplitPaneDisplay` should do the check written here, and the `InlinePhantomDisplay` can just return False.

Then these lines could be written `if worker.display.owns_view(self.view): ...`, removing the need to monkey around in the private attributes of `worker.display`.

This PR is a great change though---I'm happy to merge it right away and impose my draconian coding conventions later if you're tight on time.

Reply:
Yeah to be fair that check is quite abusive. Not sure why I went for that, maybe I didn't dare add a method at the time. Anyway I'd rather do a cleaning pass than slowly degrade the structure, so I'm happy to adjust. ^^